drivers/rtc/rtc_rv8263.c

```diff
@@ -0,0 +1,690 @@
+        timeptr->tm_hour = value[4];
+        (*p_mask) |= RTC_ALARM_TIME_MASK_WEEKDAY;
+    }
+
+    return 0;
+}
+
+static int rv8263c8_alarm_set_callback(const struct device *dev, uint16_t id,
+                                       rtc_alarm_callback callback, void *p_user_data)
+{
+    int err;
+    struct rv8263c8_data *const data = dev->data;
+    const struct rv8263c8_config *config = dev->config;
+
+    if ((dev == NULL) || (id != 0)) {
+        return -EINVAL;
+    }
+
+    (void)k_sem_take(&data->lock, K_FOREVER);
+
+    data->alarm_cb = callback;
+    data->alarm_cb_data = p_user_data;
+
+    if (callback != NULL) {
+        err = i2c_reg_update_byte_dt(&config->i2c_bus, RV8263C8_REGISTER_CONTROL_2, RV8263C8_BM_ALARM_INT_ENABLE,
+                                     RV8263C8_BM_ALARM_INT_ENABLE);
+    } else {
+        err = i2c_reg_update_byte_dt(&config->i2c_bus, RV8263C8_REGISTER_CONTROL_2, RV8263C8_BM_ALARM_INT_ENABLE,
+                                     RV8263C8_BM_ALARM_INT_DISABLE);
+    }
+
+    k_sem_give(&data->lock);
+
+    if (err < 0) {
+        LOG_ERR("Error while writing CONTROL2! Error: %i", err);
+        return err;
+    }
+
+    return 0;
+}
+
+static int rv8263c8_alarm_is_pending(const struct device *dev, uint16_t id)
+{
+    int ret;
+    struct rv8263c8_data *const data = dev->data;
+
+    if ((dev == NULL) || (id != 0)) {
+        return -EINVAL;
+    }
+
+    (void)k_sem_take(&data->lock, K_FOREVER);
+    ret = data->is_alarm_pending ? 1 : 0;
+    data->is_alarm_pending = false;
+    k_sem_give(&data->lock);
+
+    return ret;
+}
+#endif
+
+#if CONFIG_RTC_UPDATE && DT_ANY_INST_HAS_PROP_STATUS_OKAY(int_gpios)
+int rv8263_update_callback(const struct device *dev, rtc_update_callback callback, void *p_user_data)
+{
+    struct rv8263c8_data *const data = dev->data;
+
+    if (dev == NULL) {
+        return -EINVAL;
+    }
+
+    (void)k_sem_take(&data->lock, K_FOREVER);
+    data->update_cb = callback;
+    data->update_cb_data = p_user_data;
+    k_sem_give(&data->lock);
+
+    return rv8263c8_update_enable_timer(dev);
+}
+#endif
+
+#ifdef CONFIG_RTC_CALIBRATION
+int rv8263c8_calibration_set(const struct device *dev, int32_t calibration)
+{
+    int err;
+    int8_t reg;
+    struct rv8263c8_data *const data = dev->data;
+    const struct rv8263c8_config *config = dev->config;
+
+    if (dev == NULL) {
+        return -EINVAL;
+    }
+
+    return 0;
+}
+
+int rv8263c8_calibration_get(const struct device *dev, int32_t *calibration)
+{
+    int err;
+    uint8_t value;
+    struct rv8263c8_data *const data = dev->data;
+    const struct rv8263c8_config *config = dev->config;
+
+    if ((dev == NULL) || (calibration ==  NULL)) {
+        return -EINVAL;
+    }
+
+    return 0;
+}
+#endif
+
+static const struct rtc_driver_api rv8263c8_driver_api = {
+    .set_time = rv8263c8_time_set,
+    .get_time = rv8263c8_time_get,
+#if CONFIG_RTC_ALARM && DT_ANY_INST_HAS_PROP_STATUS_OKAY(int_gpios)
```

Comment:
I think `DT_ANY_INST_HAS_PROP_STATUS_OKAY(int_gpios)` is only required for the `alarm_set_callback`. The rest of the alarm interface does not require interrupts.

Reply:
I´m relying on int-gpios for the `update` API too